pkg/controller/changetierrequest/changetierrequest_controller_test.go

```diff
@@ -0,0 +1,345 @@
+package changetierrequest
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/configuration"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	murtest "github.com/codeready-toolchain/toolchain-common/pkg/test/masteruserrecord"
+	templatev1 "github.com/openshift/api/template/v1"
+	"github.com/spf13/cast"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	apiv1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestChangeTierSuccess(t *testing.T) {
+	// given
+	restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_DURATION_BEFORE_CHANGE_REQUEST_DELETION", "10s")
+	defer restore()
+	teamTier := NewNSTemplateTier("team", "123team", "stage", "dev")
+	basicTier := NewNSTemplateTier("basic", "123abc", "code", "stage", "dev")
+
+	t.Run("the controller should change tier in MUR", func(t *testing.T) {
+		// given
+		mur := murtest.NewMasterUserRecord("john")
+		changeTierRequest := newChangeTierRequest("john", "team", "", noStatus)
+		controller, request, cl := newController(t, changeTierRequest, mur, teamTier)
+
+		// when
+		_, err := controller.Reconcile(request)
+
+		// then
+		require.NoError(t, err)
+		murtest.AssertThatMasterUserRecord(t, "john", cl).
+			AllUserAccountsHaveTier(*teamTier)
+		AssertThatChangeTierRequestHasCondition(t, cl, changeTierRequest.Name, toBeComplete())
+	})
+
+	t.Run("the controller should change tier in all UserAccounts in MUR", func(t *testing.T) {
+		// given
+		mur := murtest.NewMasterUserRecord("johny", murtest.AdditionalAccounts("another-cluster"))
+		changeTierRequest := newChangeTierRequest("johny", "team", "", noStatus)
+		controller, request, cl := newController(t, changeTierRequest, mur, teamTier)
+
+		// when
+		_, err := controller.Reconcile(request)
+
+		// then
+		require.NoError(t, err)
+		murtest.AssertThatMasterUserRecord(t, "johny", cl).
+			AllUserAccountsHaveTier(*teamTier)
+		AssertThatChangeTierRequestHasCondition(t, cl, changeTierRequest.Name, toBeComplete())
```

Comment:
what about having the same fluent API that we use for other CRDs ? eg, something like: 
```
AssertThatChangeTierRequest(t, cl, changeTierRequest.Name).HasCondition(Complete())
```

Reply:
It would make sense if we had more assertions to cover with this fluent API - as we have in case of MUR or other resources. But this is not the case of `ChangeTierRequest` - I guess that it's not really worth it creating the struct and all methods to achieve the fluent API.

Reply:
Sure, but it's also a matter of consistency in our assertions